xgcm/test/test_metrics.py

```diff
@@ -311,28 +307,71 @@ def test_set_metric():
         assert k in grid_manual._metrics.keys()
 
         for metric_expected, metric in zip(v, grid_manual._metrics[k]):
-            xr.testing.assert_equal(metric_expected.reset_coords(drop=True), metric)
+            xr.testing.assert_allclose(metric_expected.reset_coords(drop=True), metric)
 
         for metric_expected, metric in zip(v, grid._metrics[k]):
-            xr.testing.assert_equal(metric_expected.reset_coords(drop=True), metric)
+            xr.testing.assert_allclose(metric_expected.reset_coords(drop=True), metric)
 
 
-@pytest.mark.skip(reason="For the next PR")
 @pytest.mark.parametrize(
-    "metric_axes,metric_name",
+    "metric_axes, exist_metric_varname, add_metric_varname, expected_varname",
     [
-        ("X", "dx_t"),
-        ("Y", "dy_ne"),
+        (
+            "X",
+            ["dx_t", "dx_n", "dx_e", "dx_ne"],
+            ["dx_n_overwrite"],
+            ["dx_t", "dx_n_overwrite", "dx_e", "dx_ne"],
+        ),
+        (
+            ("Y", "X"),
+            ["area_t", "area_n", "area_e", "area_ne"],
+            ["area_n_overwrite"],
+            ["area_t", "area_n_overwrite", "area_e", "area_ne"],
+        ),
+        # overwrite 1 existing metric, append 1 new metric
+        (
+            "X",
+            ["dx_t", "dx_n", "dx_e"],
+            ["dx_n_overwrite", "dx_ne"],
+            ["dx_t", "dx_n_overwrite", "dx_e", "dx_ne"],
+        ),
     ],
 )
-def test_interp_metrics(metric_axes, metric_name):
-    # need to test with boundary and fill_value conditions
+def test_set_metric_overwrite_true(
+    metric_axes, exist_metric_varname, add_metric_varname, expected_varname
+):
     ds, coords, metrics = datasets_grid_metric("C")
-    grid = Grid(ds, coords=coords)
-    grid.set_metrics(metric_axes, metric_name)
-    interp_metric = grid._interp_metric(ds.u, metric_axes)
 
-    expected_metric = grid.interp(ds[metric_name], metric_axes)
+    ds = ds.assign_coords({add_metric_varname[0]: ds[exist_metric_varname[1]] * 10})
+
+    metrics = {
+        k: [m for m in v if m in exist_metric_varname] for k, v in metrics.items()
+    }
+    grid = Grid(ds, coords=coords, metrics=metrics)
+
+    for av in add_metric_varname:
+        grid.set_metrics(metric_axes, av, overwrite=True)
+
+    key = frozenset(list(metric_axes))
+    set_metric = grid._metrics.get(key)
+
+    expected_metric = []
+    for ev in expected_varname:
+        metric_var = ds[ev].reset_coords(drop=True)
+        expected_metric.append(metric_var)
+
+    assert len(set_metric) == len(expected_metric)
+
+    for i in range(len(set_metric)):
+        assert set_metric[i].equals(expected_metric[i])
+
+
+def test_set_metric_overwrite_false():
```

Comment:
```suggestion
def test_set_metric_errors():
```

We should also test `grid.set_metrics("X", "foo")` raises an error.

What should `grid.set_metrics("X", "dx_e", overwrite=False)` do? It's identical to an already set metric.